search-java/src/main/java/uk/os/elements/search/android/providers/addresses/GeoPattern.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2016 Ordnance Survey
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.os.elements.search.android.providers.addresses;
+
+import java.util.regex.Pattern;
+
+final class GeoPattern {
+
+    private GeoPattern() { }
+
+    private static final Pattern UPRN_PATTERN = Pattern.compile("(^\\d{1,12}$)");
```

Comment:
Probably worth noting that these are derived form what the back end will accept.